subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h

```diff
@@ -22,45 +16,46 @@ static inline void hal_swi_init(void)
 #define HAL_SWI_JOB_IRQ    SWI5_IRQn
 #endif
 
-static inline void hal_swi_lll_pend(void)
-{
-	NVIC_SetPendingIRQ(HAL_SWI_RADIO_IRQ);
-}
+#elif defined(CONFIG_SOC_SERIES_NRF53X)
 
-static inline void hal_swi_worker_pend(void)
-{
-	NVIC_SetPendingIRQ(HAL_SWI_WORKER_IRQ);
-}
+#if defined(CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET)
 
-static inline void hal_swi_job_pend(void)
-{
-	NVIC_SetPendingIRQ(HAL_SWI_JOB_IRQ);
-}
+#define HAL_SWI_RADIO_IRQ  SWI2_IRQn
+#define HAL_SWI_WORKER_IRQ RTC0_IRQn
 
-#elif defined(CONFIG_SOC_SERIES_NRF53X)
+#if !defined(CONFIG_BT_CTLR_LOW_LAT) && \
+	(CONFIG_BT_CTLR_ULL_HIGH_PRIO == CONFIG_BT_CTLR_ULL_LOW_PRIO)
+#define HAL_SWI_JOB_IRQ    HAL_SWI_WORKER_IRQ
+#else
+#define HAL_SWI_JOB_IRQ    SWI3_IRQn
+#endif
 
-static inline void hal_swi_init(void)
-{
-	/* No platform-specific initialization required. */
-}
+#elif defined(CONFIG_BOARD_NRF5340PDK_NRF5340_CPUNET)
 
-/* Split architecture uses max. two SWI */
-#define HAL_SWI_RADIO_IRQ     EGU0_IRQn
-#define HAL_SWI_WORKER_IRQ    RTC0_IRQn
+#define HAL_SWI_RADIO_IRQ  EGU0_IRQn
+#define HAL_SWI_WORKER_IRQ RTC0_IRQn
 
 #if !defined(CONFIG_BT_CTLR_LOW_LAT) && \
 	(CONFIG_BT_CTLR_ULL_HIGH_PRIO == CONFIG_BT_CTLR_ULL_LOW_PRIO)
 #define HAL_SWI_JOB_IRQ    HAL_SWI_WORKER_IRQ
+#else
+#error "Use an unused IRQ line to implement a second SW IRQ."
+#endif
+
+#endif /* CONFIG_BOARD_NRF5340PDK_NRF5340_CPUNET */
+
+#endif /* CONFIG_SOC_SERIES_NRF53X */
+
+static inline void hal_swi_init(void)
+{
+	/* No platform-specific initialization required. */
+}
 
 static inline void hal_swi_lll_pend(void)
```

Comment:
There's a functional change, here, now: hal_swi_lll_pend() is defined unconditionally, while earlier, it was defined only under

#if !defined(CONFIG_BT_CTLR_LOW_LAT) && \
	(CONFIG_BT_CTLR_ULL_HIGH_PRIO == CONFIG_BT_CTLR_ULL_LOW_PRIO)

@cvinayak please, clarify why this change was necessary (especially for the PDK where we stick to the previous solution of using the EGU)

Reply:
`hal_swi_lll_pend()` was repeated in line 25, which is unnecessary when only the macro `HAL_SWI_RADIO_IRQ` needs conditional compilation.

Reply:
ok, i see.